core/src/main/java/org/web3j/ens/EnsResolver.java

```diff
@@ -150,6 +154,31 @@ protected OffchainResolverContract obtainOffchainResolver(String ensName) {
         }
     }
 
+    protected OffchainResolverContract obtainOffchainResolver(
+            String ensName, Credentials credentials) {
+        if (isValidEnsName(ensName, addressLength)) {
+            boolean isSynced;
+
+            try {
+                isSynced = isSynced();
+            } catch (Exception e) {
+                throw new EnsResolutionException("Unable to determine sync status of node", e);
+            }
+
+            if (!isSynced) {
+                throw new EnsResolutionException("Node is not currently synced");
+            }
+
+            try {
+                return lookupOffchainResolver(ensName, credentials);
```

Comment:
use only one try block as you catch a generic exception and throw the same one

Reply:
also the code should be a bit refactored to have only one line with "throw new"

Reply:
I have taken same logic as used before, just overloading the function with credentials param.
Shall I update the logic for both the functions?
I dont think its an ideal thing, current logic makes more sense to me, checkout this - https://docs.web3j.io/4.11.0/advanced/ethereum_name_service/#web3j-implementation